### PR TITLE
[catalyst][addressbookui] Remove AddressBookUI framework that is not available on Catalyst

### PR DESCRIPTION
**Pull request by spouliot**
This required working around internal ctor that uses internal (non
re-generated) types. E.g.

```
build/dotnet/maccatalyst/ref/Xamarin.iOS.cs(9201,64): error CS0234: The type or namespace name 'ABFunc<>' does not exist in the namespace 'AddressBookUI' (are you missing an assembly reference?)
```

We cannot just skip such constructor since the C# compiler will add a
**public** default one when none is present (and that's the case for
the error above),

This also required fixing the ctor chaining code (which can't be shared
with the, previously fixed, override code) so base classes could be
called for some `*EventArg` types.

Follow up to https://github.com/xamarin/xamarin-macios/pull/10658 , https://github.com/xamarin/xamarin-macios/pull/10678 and https://github.com/xamarin/xamarin-macios/pull/10706

This is the last framework that needs removal - but there's still a few
`unknown-*` entries in Catalyst-supported frameworks (more PR coming).

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]

File: tests/linker/ios/link sdk/LinkSdkRegressionTest.cs [hunk 1]
```diff
@@ -22,8 +22,10 @@
 using Mono.Data.Sqlite;
 #endif
 using MonoTouch;
-#if !__TVOS__ && !__WATCHOS__
+#if HAS_ADDRESSBOOKUI
```
Comment (rolfbjarne) [inline tests/linker/ios/link sdk/LinkSdkRegressionTest.cs:25]:
```suggestion
#if HAS_ADDRESSBOOK
```

File: tests/monotouch-test/AddressBook/AddressBookTest.cs [hunk 2]
```diff
@@ -7,7 +7,7 @@
 // Copyright 2012 Xamarin Inc. All rights reserved.
 //
 
-#if !__TVOS__ && !__WATCHOS__ && !MONOMAC
+#if HAS_ADDRESSBOOKUI
```
Comment (rolfbjarne) [inline tests/monotouch-test/AddressBook/AddressBookTest.cs:10]:
```suggestion
#if HAS_ADDRESSBOOK
```

File: tests/monotouch-test/ObjCRuntime/RegistrarTest.cs [hunk 3]
```diff
@@ -7,8 +7,10 @@
 
 using CoreFoundation;
 using MapKit;
-#if !__TVOS__ && !__WATCHOS__ && !MONOMAC
+#if HAS_ADDRESSBOOKUI
```
Comment (rolfbjarne) [inline tests/monotouch-test/ObjCRuntime/RegistrarTest.cs:10]:
```suggestion
#if HAS_ADDRESSBOOK
```

File: tests/linker/ios/link sdk/LinkSdkRegressionTest.cs [hunk 4]
```diff
@@ -172,7 +174,7 @@ public void Bug928_MonoModule_LinkedOut ()
 			Assert.That (m.GetType ().Name, Is.EqualTo ("RuntimeModule"), "RuntimeModule");
 		}
 
-#if !__TVOS__ && !__WATCHOS__
+#if HAS_ADDRESSBOOKUI
```
Comment (rolfbjarne) [inline tests/linker/ios/link sdk/LinkSdkRegressionTest.cs:177]: Does this needs a dual condition?

```suggestion
#if HAS_ADDRESSBOOK && HAS_ADDRESSBOOKUI
```

File: tests/monotouch-test/ObjCRuntime/RegistrarTest.cs [hunk 5]
```diff
@@ -2202,7 +2204,7 @@ public void WillEndDragging (UIScrollView scrollView, PointF velocity, ref Point
 		}
 #endif // !__WATCHOS__
 
-#if !__TVOS__ && !__WATCHOS__ && !MONOMAC// No ABPeoplePickerNavigationControllerDelegate
+#if HAS_ADDRESSBOOKUI
```
Comment (rolfbjarne) [inline tests/monotouch-test/ObjCRuntime/RegistrarTest.cs:2207]: Same, dual condition?

```suggestion
#if HAS_ADDRESSBOOK && HAS_ADDRESSBOOKUI
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

**Comment by rolfbjarne** [comment]
Should the frameworks you've been removed be removed here too: https://github.com/xamarin/xamarin-macios/blob/26c969c40622128b55f5fa2b87f938abccd931b9/tools/common/Frameworks.cs#L582?

File: tests/linker/ios/link sdk/LinkSdkRegressionTest.cs (hunk 1 above)
Comment (spouliot) [reply tests/linker/ios/link sdk/LinkSdkRegressionTest.cs:25]: weird, looks like some code completion gone bad

File: tests/linker/ios/link sdk/LinkSdkRegressionTest.cs (hunk 4 above)
Comment (spouliot) [reply tests/linker/ios/link sdk/LinkSdkRegressionTest.cs:177]: I took a shortcut as `AddressBookUI` implies `AddressBook` but there's no harm in being explicit

**Comment by spouliot** [comment]
> Should the frameworks you've been removing be removed here too:

yes. I'll do that in a different PR, double checking all entries (and adjusting tests if needed)

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**